contracts/Auction.sol

```diff
@@ -2,75 +2,145 @@
+    // (X) - Address owner_address - current owner
+    // (X) - Float price - minimum_price/marked_by_seller
+    // (Return type is different) - Float current_highest_bid - the highest bid at the time
+    // (X) - Address current_highest_bidder - the highest bidder at the time
+    //     - List [] bid_history - bids that have been placed with addresses and bid amount
+    //    - Time&Date Start_time - time of start.
+    // (Return type is different) - Time&Date End_time - time the contract is supposed to end
+    // (same as owner_address right?) - Ethereum address - Auction’s owner’s address
+    // (is basically current_highest_bidder after the auction is over right?) - Ethereum address - Item’s owner address
+    //     - MinimumBidIncrement - minimum amount to bid (set to 1% of the item's current price)
 
-  }
 
 
-   struct item {
-     uint id;
-     string name;
-   }
 
+    address payable public seller_address;
+    uint public End_time;
+    unit public price;
 
-  function viewItem(item memory x) public {
+    // Current state of the auction.
+    address public current_highest_bidder;
+    uint public current_highest_bid;
 
-  }
+    // Set to true at the end, disallows any change.
+    // By default initialized to `false`.
+    bool ended;
 
+    constructor() public {
+        
+    }
 
-  function getName(item x) public {
+    constructor(
+        uint auction_Duration,
+        address payable _seller_address
+        uint starting_price
+    ) public {
+        seller_address = _seller_address;
+        End_time = now + auction_Duration;
+        price = starting_price;
+    }
 
-  }
 
+    struct item {
+        uint id;
+        string name;
+        string description;
+    }
 
-  function getDescription(item x) public {
+    struct bid {
+        address public bidder_address;
+        uint public bid_amount;
+        uint public Time_placed;
+    }
 
-  }
 
+    function viewItem(item memory x) public {
+        // Leads to a page of the item which includes information like name, description, time remaining, bidding history.
+        // Has helper functions.
+    }
 
-  function getTimeRemaining(item x) public {
 
-  }
+    function getName(item x) public {
+        // Function: Returns the name of the item.
 
+        return x.name;
+    }
 
-  function getBiddingHistory(item x) public {
-    //Displays bidding history which is a list of outbidded prices with timestamps.
-  }
 
+    function getDescription(item x) public {
+        // Function: Returns description of the item.
 
-  function transferOwnership(address x, address y, item x) public {
+        return x.description;
+    }
 
-  }
 
+    function getTimeRemaining(item x) public {
+        // Condition(s): Auction cannot be over.
+        // Function: Returns time remaining in the auction.
 
-  function sendFunds(address x, address y, fixed128x10 cost) public {
+        require(
+            now <= auctionEndTime,
```

Comment:
`auction_end_time`

Reply:
this is actually an accident where it was mean to be end_time but I gave the wrong name for the variable